From c0f3d7726fa07a955608deb465453af17b0a383f Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Tue, 26 Jun 2018 16:32:39 +0200
Subject: [PATCH] libril: Bounds check s_commands

* We shouldn't be assuming this is always going to be valid

Change-Id: Ibd43133340d86f7c60ead5f7de63c73b05d00f2e
(cherry picked from commit f2d2baba33fe33b8b065df1986f999921eaf2c0f)
(cherry picked from commit f64c3605ebdc5a1a0cd020435f9f079b3dfebcd8)
---
 libril/ril.cpp | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libril/ril.cpp b/libril/ril.cpp
index 074a858..f31076d 100644
--- a/libril/ril.cpp
+++ b/libril/ril.cpp
@@ -223,6 +223,11 @@ addRequestToList(int serial, int slotId, int request) {
 #endif
 #endif
 
+    if (request >= (int)NUM_ELEMS(s_commands)) {
+        RLOGE("Request %s not supported", requestToString(request));
+        return NULL;
+    }
+
     pRI = (RequestInfo *)calloc(1, sizeof(RequestInfo));
     if (pRI == NULL) {
         RLOGE("Memory allocation failed for request %s", requestToString(request));
